bundle/config/validate/required.go

```diff
@@ -82,3 +83,52 @@ func (f *required) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics
 
 	return diags
 }
+
+// Bespoke code to error for fields that are not marked as required in the Go SDK / OpenAPI spec.
+func errorForMissingFields(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	// Dashboards should always have a name and warehouse_id.
+	var nameLocations []dyn.Location
+	var namePaths []dyn.Path
+	var warehouseIdLocations []dyn.Location
+	var warehouseIdPaths []dyn.Path
+
+	diags := diag.Diagnostics{}
+	for key, dashboard := range b.Config.Resources.Dashboards {
+		if dashboard.DisplayName == "" {
```

Comment:
This evaluates to true both for nil and "", are both disallowed?

If you have a link a doc where it says would be good to add it here.

Reply:
Yeah, you get an server side error:
```
Field 'display_name' is required, expected non-default value (not "")!
```

I don't think this is documented somewhere, this is just the behaviour for required fields where zero and nil are same for the server.

Reply:
Worth including this context in the comment in the code.